mlflow/sentence_transformers/__init__.py

```diff
@@ -371,18 +402,58 @@ def predict(self, sentences, params: Optional[Dict[str, Any]] = None):
 
         :return: Model predictions.
         """
-        # When the input is a single string, it is transformed into a DataFrame with one column
-        # and row, but the encode function does not accept DataFrame input
+        # When the input is a single string or a dictionary, it is transformed into a DataFrame with one column
+        # and row. When the input is, but the encode function does not accept DataFrame input
+        convert_output = False
         if type(sentences) == pd.DataFrame:
-            sentences = sentences[0]
+            # Wrap the output to OpenAI format only when the input is dict `{"input": ... }`
+            if list(sentences.columns)[0] == "input":
+                convert_output = True
+            sentences = sentences.iloc[0, 0]
+            if type(sentences) == str:
+                sentences = [sentences]
 
         # The encode API has additional parameters that we can add as kwargs.
         # See https://www.sbert.net/docs/package_reference/SentenceTransformer.html#sentence_transformers.SentenceTransformer.encode
         if params:
             try:
-                return self.model.encode(sentences, **params)
+                output_data = self.model.encode(sentences, **params)
             except TypeError as e:
                 raise MlflowException.invalid_parameter_value(
                     "Received invalid parameter value for `params` argument"
                 ) from e
-        return self.model.encode(sentences)
+        else:
+            output_data = self.model.encode(sentences)
+
+        if convert_output:
+            output_data = self.postprocess_output_for_embedding_task(sentences, output_data)
+        return output_data
+
+    def postprocess_output_for_embedding_task(
+        self, data: Union[str, List[str]], output_tensers: List[List[int]]
+    ):
+        """
+        Wrap output data with usage information.
+        :param data: text input prompts
+        :param output_tensers: List of output tensors that contain the generated embeddings
+        :return: Dictionaries containing the output embedding and usage information for each input prompt.
+        """
+        output_dict = {"object": "list"}
+        output_data = []
+        for idx, output_tensor in enumerate(output_tensers):
```

Comment:
```suggestion
        output_dict["data"] = [{
            "object": "embedding",
            "index": i,
            "embedding": tensor
         } for i, tensor in enumerate(output_tensors)]
```

nit: let's use list comprehension to be more concise and simple.